indico/web/client/js/react/components/DateRangePicker.jsx

```diff
@@ -0,0 +1,224 @@
+// This file is part of Indico.
+// Copyright (C) 2002 - 2024 CERN
+//
+// Indico is free software; you can redistribute it and/or
+// modify it under the terms of the MIT License; see the
+// LICENSE file for more details.
+
+import moment from 'moment';
+import PropTypes from 'prop-types';
+import React from 'react';
+
+import DatePickerCalendar from 'indico/react/components/DatePickerCalendar';
+import {FinalField, validators as v} from 'indico/react/forms';
+import {Param, Translate} from 'indico/react/i18n';
+import {DateRange} from 'indico/utils/date';
+import {formatDate} from 'indico/utils/date_format';
+import {fromISOLocalDate} from 'indico/utils/date_parser';
+
+import 'indico/custom_elements/ind_date_picker';
+
+const INVALID = '__invalid__';
+
+export default function DateRangePicker({
+  value,
+  label,
+  format,
+  rangeStartLabel,
+  rangeEndLabel,
+  rangeStartMin,
+  rangeStartMax,
+  rangeEndMin,
+  rangeEndMax,
+  readOnly,
+  disabled,
+  onChange,
+}) {
+  format ??= moment.localeData().longDateFormat('L');
+
+  function handleChange(ev) {
+    const picker = ev.currentTarget;
+    const [start, end] = picker.dateRange;
+    const rangeStartInput = picker.querySelector('[data-range-start]');
+    const rangeEndInput = picker.querySelector('[data-range-end]');
+
+    const startInvalid = rangeStartInput.value && !start;
+    const endInvalid = rangeEndInput.value && !end;
+
+    onChange({
+      // ISO local dates
+      startDate: startInvalid ? INVALID : start?.toLocaleDateString('en-ca') || null,
+      endDate: endInvalid ? INVALID : end?.toLocaleDateString('en-ca') || null,
+    });
+  }
+
+  return (
+    <ind-date-range-picker
+      range-start={fromISOLocalDate(value.startDate)?.toDateString() || ''}
+      range-end={fromISOLocalDate(value.endDate)?.toDateString() || ''}
+      range-start-min={rangeStartMin}
+      range-start-max={rangeStartMax}
+      range-end-min={rangeEndMin}
+      range-end-max={rangeEndMax}
+      onChange={handleChange}
+    >
+      <fieldset>
+        <legend>{label}</legend>
+        <label>
+          <span>{rangeStartLabel}:</span>
+          <input
+            type="text"
+            data-range-start
+            placeholder={format}
+            readOnly={readOnly}
+            disabled={disabled}
+          />
+        </label>
+        <span className="arrow" />
+        <label>
+          <span>{rangeEndLabel}:</span>
+          <input
```

Comment:
After selecting the date range in the popup, it auto-selects the end date. This feels rather weird and I guess it's not intentional?

![image](https://github.com/user-attachments/assets/55d6bf93-24eb-49c0-a31c-2c20a64fdcb6)

Reply:
Now that you mention it, it makes more sense to focus the button when operating this with the keyboard.